hoomd/hpmc/HPMCPrecisionSetup.h

```diff
@@ -23,31 +23,9 @@ namespace hoomd
     {
 namespace hpmc
     {
-#ifdef SINGLE_PRECISION
-
-// in single precision, OverlapReal is always float
-//! Typedef'd real for use in local overlap checks
-typedef float OverlapReal;
-//! Typedef'd real3 for use in the local overlap checks
-typedef float3 OverlapReal3;
-typedef float4 OverlapReal4;
-
-#else
-
-// in double precision, mixed mode enables floats for OverlapReal, otherwise it is double
-#ifdef ENABLE_HPMC_MIXED_PRECISION
-typedef float OverlapReal;
-typedef float3 OverlapReal3;
-typedef float4 OverlapReal4;
-
-#else
-typedef double OverlapReal;
-typedef double3 OverlapReal3;
-typedef double4 OverlapReal4;
-
-#endif
-
-#endif
+typedef ShortReal OverlapReal;
+typedef ShortReal3 OverlapReal3;
+typedef ShortReal4 OverlapReal4;
```

Comment:
Is there a reason to keep this typedef?

Reply:
To temporarily avoid changing 3037 instances of `OverlapReal` in 47 files. We don't know how long until the v4 release, so `trunk-major` will be a long lived branch. Better to not force merge conflicts on any upcoming HPMC development in the 3.x series. We can make this change just prior to a v4.0.0 release.

Reply:
#1447 reminds us to make this change.